src/pymmcore_plus/mda/handlers/_ome_zarr_writer.py

```diff
@@ -228,8 +228,11 @@ def frameReady(
 
             # create the new array, getting XY chunksize from the frame
             # and total shape from the sequence
-            shape = (*tuple(v for k, v in seq.sizes.items() if k != "p"), *frame.shape)
-            axes = (*(k for k in seq.sizes if k != "p"), "y", "x")
+            shape = (
+                *tuple(v for k, v in seq.sizes.items() if k != "p" and v > 0),
+                *frame.shape,
+            )
+            axes = (*(k for k, v in seq.sizes.items() if k != "p" and v > 0), "y", "x")
```

Comment:
```suggestion
            axes = (*(k for k in seq.used_axes if k != "p"), "y", "x")
            shape = (*tuple(v for k, v in seq.sizes.items() if k in axes), *frame.shape)
```

how about this?  There's an attribute called `used_axes` that takes core of these "empty" axes.